### PR TITLE
chore: migrate ImageGallery to packages

### LINKED ISSUE #1723: src/components/ImageGallery/ImageGallery.tsx
We are looking to migrate `src/components/ImageGallery/ImageGallery.tsx` to `packages/components/src/ImageGallery/ImageGallery.tsx`

**Why?**

-   Improve structure of the components
-   Improve documentation for our existing components
-   Make it easier for our design team to review and refine the components
-   Decouple our components from application state
-   Convert our class based components to functional components
-   Allow reuse of our components outside of the primary single page application (SPA)#

**Definition of Migrated**

-   The component file exists in `packages/components/src`
-   There is a companion `*.stories.tsx` which includes an example for each use case of the component

**How?**

1.  [Create a new component](https://github.com/ONEARMY/community-platform/tree/master/packages/components#creating-a-new-component) in `./packages/components`
2.  Move existing component into new component
3.  Update all references to import the new component from `oa-components`
4.  Delete all files from `./src/components`

🐉 **Things to watch out for:**

-   If the original component file exported _multiple_ components these should be split into multiple separate files
-   If the component you're working on imports another component from `src/components` you will need to migrate that item first. Please make sure to claim ownership on the associated issue.
-   We want to use functional components, [read more around how to refactor a class component](https://nimblewebdeveloper.com/blog/convert-react-class-to-function-component)
-   You may spot an opportunity to create a new component to encapsulate some elements which are being used frequently across the UI. Yay, a new component is born 🐥
- The component may not need to exist.

### PR DESCRIPTION
**Pull request by Xyli0**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Developer experience (improves developer workflows for contributing to the project)

## Description

Migrates component `ImageGallery.tsx` from `/src/components/ImageGallery` to `/packages/components/src/ImageGallery` and  `.stories.tsx` file.

The story loads images of different sizes as placeholders from the `https://picsum.photos/` API.

## Git Issues

Closes #1723 

## Screenshots/Videos
Screenshot 2022-06-29 at 18-09-32 Base Components _ ImageGallery - Default ⋅ Storybook

### CODE REVIEW

File: packages/components/src/ImageGallery/ImageGallery.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+const imageUrls = [
+  'https://picsum.photos/1500/500',
+  'https://picsum.photos/4000/3000',
+  'https://picsum.photos/800/1200',
+  'https://picsum.photos/1500/1500',
+]
+
+const testImages: IUploadedFileMeta[] = imageUrls.map((elt, i) => {
```
Comment (Xyli0) [inline packages/components/src/ImageGallery/ImageGallery.stories.tsx:22]: As said above here the type is checked just to show that it is conform to what's awaited but not mandatory.

**Comment by Xyli0** [comment]
Hi! 
As just reviewed the main problem is that the component can't access the `src/stores/storage` file so I basically copied its `IUploadedMeta` type into the component. 
So the component works but it could be more DRY I guess.
If this implementation works for now i'll change it to PR but I wanted your thoughts about it.
Thanks.

**Review by thisislawatts — APPROVED** [review]

**Comment by thisislawatts** [comment]
This looks great [@Xyli0](https://github.com/Xyli0), thanks for the time and thought on this so far. I appreciate your concern around the `IUploadedFileMeta`, however I think it's preferable to have interfaces clearly defined at each layer. For example now if the data structure changes within the database we should be free to introduce either an adapter before the data reaches the UI **or** refactor the UI layer. 

We did something similar in the [CommentItem]
(https://github.com/ONEARMY/community-platform/blob/master/packages/components/src/CommentItem/CommentItem.tsx#L9-L21) althought it would be good to refactor that a little further to rename the properties like `_id` to `id`, this removes a stylistic detail of the firebase documents underpinning the UI. 

Happy to merge this as is, I've converted this to a standard PR, if you can update the branch we will be good to merge ✨

**Comment by Xyli0** [comment]
@thisislawatts thank you for your review and for your complete answer, it helps to understand more deeply the code base and the choices that have been made, it's really appreciated! 
I just resolved the conflicts on the branch.
Thanks a lot.

**Review by thisislawatts — APPROVED** [review]

**Pull request merged**